src/main/java/bio/terra/dao/AssetDao.java

```diff
@@ -126,31 +126,30 @@ public void retrieve(Study study) {
                                                                  Map<UUID, StudyTable> allTables,
                                                                  Map<UUID, StudyTableColumn> allColumns) {
         Map<UUID, AssetTable> tables = new HashMap<>();
-        List<Map<String, Object>> results = jdbcTemplate.queryForList(
-                "SELECT asset_column.id, asset_column.study_column_id, study_column.table_id " +
-                        "FROM asset_column " +
-                        "INNER JOIN study_column ON asset_column.study_column_id = study_column.id " +
-                        "WHERE asset_id = :assetId",
-                new MapSqlParameterSource().addValue("assetId", spec.getId()));
-        results
-                .forEach(rs -> {
-                    UUID tableId = UUID.fromString(rs.get("table_id").toString());
-                    UUID columnId = UUID.fromString(rs.get("study_column_id").toString());
-                    if (!tables.containsKey(tableId)) {
-                        tables.put(tableId, new AssetTable().studyTable(allTables.get(tableId)));
-                    }
-                    AssetTable assetTable = tables.get(tableId);
-                    AssetColumn newColumn = new AssetColumn()
-                            .id(UUID.fromString(rs.get("id").toString()))
-                            .studyColumn(allColumns.get(columnId));
-                    // check to see if this table and column are the root values
-                    if (rootTableId.equals(tableId) && rootColumnId.equals(columnId)) {
-                        spec.rootTable(assetTable);
-                        spec.rootColumn(newColumn);
-                    }
-                    // add the new column to the asset table object
-                    assetTable.getColumns().add(newColumn);
-                });
+        String sql = "SELECT asset_column.id, asset_column.study_column_id, study_column.table_id " +
+                "FROM asset_column " +
+                "INNER JOIN study_column ON asset_column.study_column_id = study_column.id " +
+                "WHERE asset_id = :assetId";
+        MapSqlParameterSource params = new MapSqlParameterSource().addValue("assetId", spec.getId());
+        List<Map<String, Object>> results = jdbcTemplate.queryForList(sql, params);
+        results.forEach(rs -> {
+            UUID tableId = (UUID) rs.get("table_id");
```

Comment:
Should these be getObject(... UUID.class) too?
I'm not sure what the best practice is.

Reply:
the variable name rs is misleading. i'm looping through a regular Map<String,Object> and map does not have a getObject method.